pkg/deviceclaimingserver/enddevices/ttjsv2/ttjs.go

```diff
@@ -322,6 +323,91 @@ func (c *TTJS) GetClaimStatus(
 	}
 }
 
+// BatchUnclaim implements EndDeviceClaimer.
+func (c *TTJS) BatchUnclaim(
+	ctx context.Context,
+	ids []*ttnpb.EndDeviceIdentifiers,
+) (map[types.EUI64]errors.ErrorDetails, error) {
+	ret := make(map[types.EUI64]errors.ErrorDetails)
+	if len(ids) == 0 {
+		return ret, nil
+	}
+	var euis string
+	joinEUI := types.MustEUI64(ids[0].JoinEui).OrZero()
+	for _, ids := range ids {
+		devEUI := types.MustEUI64(ids.DevEui).OrZero()
+		if euis != "" {
+			euis += ","
+		}
+		euis += devEUI.String()
+	}
+
+	reqURL := fmt.Sprintf("%s/api/v2/devices/%s/claims", c.config.URL, euis)
+	logger := log.FromContext(ctx).WithFields(log.Fields(
+		"join_eui", joinEUI,
+		"url", reqURL,
+	))
+
+	logger.Debug("Unclaim end devices")
+	request, err := http.NewRequestWithContext(ctx, http.MethodDelete, reqURL, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	client, err := c.httpClient(ctx)
+	if err != nil {
+		return nil, err
+	}
+	resp, err := client.Do(request)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	respBody, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	if isSuccess(resp.StatusCode) {
+		return ret, nil
+	}
+
+	switch resp.StatusCode {
+	case http.StatusBadRequest:
+		var (
+			resp    EndDevicesErrors
+			message string
+		)
+		err = json.Unmarshal(respBody, &resp)
+		if err == nil {
+			for euiText, r := range resp {
+				var eui types.EUI64
+				err := eui.UnmarshalText([]byte(euiText))
+				if err != nil {
+					logger.WithError(err).WithField("eui", euiText).Warn("Failed to decode EUI")
+					continue
+				}
+				ret[eui] = errUnclaimDevice.WithAttributes("message", r.Message)
+			}
+		} else {
+			var errResp ErrorResponse
+			err = json.Unmarshal(respBody, &errResp)
+			if err != nil {
+				logger.WithError(err).Warn("Failed to decode error message")
+			} else {
+				logger.WithField("error", errResp.Message).Warn("Failed to unclaim end devices")
+			}
+			message = errResp.Message
+		}
```

Comment:
This is how I've interpreted the `oneof` https://github.com/TheThingsIndustries/lorawan-join-server/blob/master/api/openapi.json#L495-L511
Please confirm if this is correct.